gstreamer/src/query.rs

```diff
@@ -208,17 +273,156 @@ impl fmt::Debug for QueryRef {
     }
 }
 
-impl ToOwned for QueryRef {
-    type Owned = GstRc<QueryRef>;
+impl Query {
+    pub fn new_position(fmt: ::Format) -> Position<QueryRef> {
```

Comment:
My idea here was that we put `Query` into the type parameter directly. So that there can be `Position<Query>`, `Position<&QueryRef>` and `Position<&mut QueryRef>`. The first would own the query, and own it uniquely (refcount == 1), the others borrow the query and the last one has it borrowed uniquely (refcount == 1)

Reply:
I.e. `Position<GstRc<QueryRef>>` (`QueryRef` as it was before). You can then safely deref/deref_mut the inside of that to a `&QueryRef` and `&mut QueryRef` (because you know that refcount==1). And then wrap a `Position<&mut QueryRef>` around it, which gives access to all the API. Just not sure how to implement `Deref`/`DerefMut` so that we can directly go to a `&mut Position<&mut QueryRef>`, or how we otherwise get access to all the API we care about on the `Position<Query>` in a convenient way.